std/assembly/array.ts

```diff
@@ -105,9 +105,14 @@ export class Array<T> extends ArrayBufferView {
   }
 
   @operator("[]=") private __set(index: i32, value: T): void {
-    ensureSize(changetype<usize>(this), index + 1, alignof<T>());
-    this.__unchecked_set(index, value);
-    if (index >= this.length_) this.length_ = index + 1;
+    if (index >= this.length_) {
+      ensureSize(changetype<usize>(this), index + 1, alignof<T>());
+      this.__unchecked_set(index, value);
+      this.length_ = index + 1;
+    } else {
+      if (index < 0) throw new RangeError(E_INDEXOUTOFRANGE);
```

Comment:
Wondering if we can move this check into the if branch by changing the first check to `<u32>index >= <u32>this.length_`, which I'd assume is less common so we could make the common case one less branch?

Reply:
Yeah good catch!